lib/Items/ApplicationDockItem.vala

```diff
@@ -442,11 +442,12 @@ namespace Plank
 				
 				foreach (var s in actions) {
 					var values = actions_map.get (s).split (";;");
-					
 					var item = create_menu_item (s, values[1], true);
 					item.activate.connect (() => {
 						try {
-							AppInfo.create_from_commandline (values[0], null, AppInfoCreateFlags.NONE).launch (null, null);
+							var desktop_id = File.new_for_uri (Prefs.Launcher).get_basename ();
```

Comment:
Is there a more robust way of getting the `desktop_id`? The `File.new_for_uri (Prefs.Launcher)` is used in other places in this class so I don't think it will fail, but it would be nice to get the id more directly.

Reply:
The `main` branch has a better way but this is probably OK for `master`.  You could maybe use `Path.get_basename (Prefs.Launcher)` ?

Reply:
I've updated it in [this commit](https://github.com/elementary/dock/pull/166/commits/c6b2b68245b4f89231d1fdf4601ec80da109a952). I tested the Path approach with a mix of different app actions and it works OK.